File: Cpp/1160_Find_Words_That_Can_Be_Formed_by_Characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        int result = 0;
        vector<int> frequency(26, 0);
        for(auto& ch : chars){
            ++frequency[ch-'a'];
        }
        
        for(int i=0; i<words.size(); ++i){
            vector<int> temp(26, 0);
            bool not_found = true;
            for(auto& ch : words[i]){
                ++temp[ch-'a'];
                if(temp[ch-'a']>frequency[ch-'a']) {
                    not_found = false;
                    break;
                }
            }
            
            if(not_found){
                result += words[i].size();
            }
        }
        
        return result;
    }
};

File: Kotlin/1160_Find_Words_That_Can_Be_Formed_by_Characters.kt
class Solution {
    fun countCharacters(words: Array<String>, chars: String): Int {
        var result = 0
        val frequency = IntArray(26)
        
        for(ch in chars){
            ++frequency[ch - 'a']
        }
        
        for (word in words) {
            val temp = IntArray(26)
            var notFound = true
            for(ch in word){
                ++temp[ch - 'a']
                
                if(temp[ch - 'a'] > frequency[ch - 'a']) {
                    notFound = false
                    break
                }
            }
            
            if(notFound){
                result += word.length
            }
        }
        
        return result
    }
}